src/http/headers.cr

```diff
@@ -197,33 +345,89 @@ struct HTTP::Headers
     true
   end
 
-  def each
+  # Returns an `Iterator` over the header keys. Which behaves like an `Iterator` returning a `Tuple` consisting of the header key and value.
```

Comment:
This method only yields, it does not return an iterator. Only the second half of this doc comment applies.